packages/core/src/model.ts

```diff
@@ -52,18 +52,29 @@ export class ChartModel {
 		const { groupMapsTo } = this.getOptions().data;
 
 		const axesOptions = this.getOptions().axes;
-		
+
 		// Check for custom domain
 		if (axesOptions) {
 			Object.keys(axesOptions).forEach(axis => {
-				if (axesOptions[axis].domain && axesOptions[axis].mapsTo) {
-					const mapsTo = axesOptions[axis].mapsTo;
-					const [start, end] = axesOptions[axis].domain;
-
-					// Filter out data outside domain
-					displayData = displayData.filter((datum) => 
-						datum[mapsTo] >= start && datum[mapsTo] <= end
-					)
+				const mapsTo = axesOptions[axis].mapsTo;
+
+				if (
+					mapsTo &&
+					axesOptions[axis].domain &&
+					!this.getOptions().percentage
+				) {
```

Comment:
Assuming prettier isn't formatting it like this maybe:
```suggestion
				if (mapsTo
					&& axesOptions[axis].domain
					&& !this.getOptions().percentage) {
```

Reply:
I don't think it formats in this way.

Reply:
okay